src/crimson/osd/osd_operations/pg_splitting.h

```diff
@@ -0,0 +1,58 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <iostream>
+#include <seastar/core/future.hh>
+
+#include "crimson/osd/osd_operation.h"
+#include "crimson/osd/osd_operations/peering_event.h"
+#include "osd/osd_types.h"
+#include "crimson/common/type_helpers.h"
+
+namespace ceph {
+  class Formatter;
+}
+
+namespace crimson::osd {
+
+class ShardServices;
+class PG;
+/*
+ has to happen before PGAdvanceMap 
+ we assume the pg->get_osdmap() is last_map and we will get next_map from shard_services.get_map(next_epoch)
+*/
+
+class PGSplitting : public PhasedOperationT<PGSplitting> {
+public:
+  static constexpr OperationTypeCode type = OperationTypeCode::pg_splitting;
+
+protected:
+  Ref<PG> pg;
+  ShardServices &shard_services;
+  PipelineHandle handle;
+
+  OSDMapRef cur_map;
+  epoch_t next_epoch;
```

Comment:
Can you please add an explanation comment here?
As far as I understand we've just finished PGAdvanceMap from `cur_map.epoch` up to `next_epoch` and we will be calculating any splits in that range (retrospectively), right?

Perhaps we should name it old_map and new_map?

IIUC, `next_epoch` shouldn't actually be passed to PGSplitting as this is the current map of this pg (since we just advanced up to this map earlier). We can still pass it in order to assert that this is, in face, the current map we have.